### ISSUE TITLE
Tracking issue for test runner. It is and has been failing

### ISSUE THREAD

**Issue by davethiede** [LibertyOS-Development/kernel#28]
**Describe the bug**

**To Reproduce**
1. cargo test
2.

**Expected behavior**
The tests should complete successfully and cargo test exit gracefully.

**Screenshots**

**Additional information**
 - Architecture: [e.g. x86_64, ARMv7] x86_64
 - OS: [e.g. Windows 10, Ubuntu 21.10] archlinux current
 - Build environment: [e.g. WSL2/Docker/plain linux] plain linux
 - QEMU version: [paste output from `qemu-system-x86_64 --version`]
 - rustc version: [paste output from `rustc --version`] rustc 1.73.0-nightly (0e8e857b1 2023-07-16)
 - Commit hash: [link to commit from which binary was built. **Do not paste link to main or any other branch head. Use concrete commit hash**] This has been an ongoing issue tag:v0.17.4+6

**Additional context**

**Comment by danielteberian** [comment]
Yes, I mentioned this somewhere the other day. You can run tests if you run them individually, but you cannot run all the tests with "cargo test". I would _love_ to fix this, especially because of how many tests I could add. I think that it would be a great step forward to have the ability to test all the modules and functions of the kernel. How do you think we should go about doing this?

**Comment by davethiede** [comment]
I had tried to find out how to assign this to me but didn't see the option. I'm still learning the build environment so no clear path forward. I'm trying to start with the basics to get a simple build but have been running into errors when I try to do a "cargo boot image". the errors are all related to  missing things like Ok() and Err(). it is apparent that something is missing in my simple test environment.  is there a chat channel related to this os?

**Comment by althea89sky** [comment]
hi, i am interested to build our app on your operating system, is there a possible way that we can work together? im veronica and you can reach me out in telegram veronicaandrino , i tried to find you in twitter but i cant find it, so @davethiede @Kepler-Br @danielteberian  so is it possible to meet on call to discuss what we are building? thanks.

**Comment by danielteberian** [comment]
I don't have Discord (I don't use it), but we can contact each other via Github, email, or something like that.
